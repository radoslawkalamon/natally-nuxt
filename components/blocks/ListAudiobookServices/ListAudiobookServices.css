.list-audiobook-services {
  display: flex;
  flex-flow: row wrap;
  gap: var(--gap);
  justify-content: center;
  list-style: none;
  margin: 0 auto;
  max-width: var(--max-width-body);
  padding: 0;
}

.list-audiobook-services__item {
  max-width: 260px;
  width: 100%;
}
