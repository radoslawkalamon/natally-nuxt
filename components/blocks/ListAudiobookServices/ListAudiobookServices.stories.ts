import type { Meta, Story } from '@storybook/vue'
import BlocksListAudiobookServices from '@/components/blocks/ListAudiobookServices/ListAudiobookServices.vue'

const meta: Meta = {
  title: 'Blocks / List Audiobook Services',
  component: BlocksListAudiobookServices
}
export default meta

export const Default: Story = () => ({
  components: {
    BlocksListAudiobookServices
  },
  template: '<BlocksListAudiobookServices />'
})
