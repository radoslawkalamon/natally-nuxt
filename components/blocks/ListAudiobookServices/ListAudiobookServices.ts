import Vue from 'vue'

type AudiobookService = {
  icon: String,
  name: String,
  url: String,
}

export default Vue.extend({
  name: 'BlocksListAudiobookServices',
  components: {
    ComponentsCardAudiobookService: () => import(/* webpackChunkName: "components" */'@/components/components/CardAudiobookService/CardAudiobookService.vue')
  },
  computed: {
    audiobookServices (): AudiobookService[] {
      return [
        {
          icon: require('./images/apple-podcasts.svg'),
          name: 'Apple Podcasts',
          url: 'https://169cm.pl/audiobooki/itunes'
        },
        {
          icon: require('./images/castbox.svg'),
          name: 'Castbox',
          url: 'https://169cm.pl/audiobooki/castbox'
        },
        {
          icon: require('./images/google-podcasts.svg'),
          name: 'Google Podcasts',
          url: 'https://169cm.pl/audiobooki/google-podcasts'
        },
        {
          icon: require('./images/pocket-casts.svg'),
          name: 'Pocket Casts',
          url: 'https://169cm.pl/audiobooki/pocket-casts'
        },
        {
          icon: require('./images/soundcloud.svg'),
          name: 'Soundcloud',
          url: 'https://169cm.pl/audiobooki/soundcloud'
        },
        {
          icon: require('./images/spotify.svg'),
          name: 'Spotify',
          url: 'https://169cm.pl/audiobooki/spotify'
        },
        {
          icon: require('./images/youtube.svg'),
          name: 'YouTube',
          url: 'https://169cm.pl/audiobooki/youtube'
        }
      ]
    }
  }
})
