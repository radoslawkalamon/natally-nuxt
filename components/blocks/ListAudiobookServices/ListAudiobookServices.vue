<template>
  <ul class="list-audiobook-services">
    <template v-for="(audiobookService, index) in audiobookServices">
      <li
        :key="index"
        class="list-audiobook-services__item"
      >
        <ComponentsCardAudiobookService v-bind="audiobookService" />
      </li>
    </template>
  </ul>
</template>

<script src="./ListAudiobookServices.ts" lang="ts"></script>
<style src="./ListAudiobookServices.css" scoped></style>
