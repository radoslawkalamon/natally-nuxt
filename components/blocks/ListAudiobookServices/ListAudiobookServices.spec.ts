import ListAudiobookServices from '@/components/blocks/ListAudiobookServices/ListAudiobookServices.vue'
import { shallPassIntegrationSanityTest, shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

describe('Blocks / List Audiobook Services', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: ListAudiobookServices,
      options: {
        stubs: [
          'ComponentsCardAudiobookService'
        ]
      }
    })
  })

  describe('Integration', () => {
    shallPassIntegrationSanityTest({ component: ListAudiobookServices })
  })
})
