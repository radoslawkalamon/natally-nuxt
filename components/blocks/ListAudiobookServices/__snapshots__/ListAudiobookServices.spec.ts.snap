// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks / List Audiobook Services Integration shall pass integration sanity test 1`] = `
"<ul data-test=\\"blocks-list-audiobook-services\\" class=\\"list-audiobook-services\\">
  <li class=\\"list-audiobook-services__item\\"><a data-test=\\"components-card-audiobook-service\\" href=\\"https://169cm.pl/audiobooki/itunes\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"card-audiobook-service\\">
      <div class=\\"card-audiobook-service__logo-wrapper\\"><img alt=\\"Ikona serwisu Apple Podcasts\\" data-test=\\"components-card-audiobook-service-logo\\" src=\\"apple-podcasts.svg\\" class=\\"card-audiobook-service__logo\\"></div>
      <div class=\\"card-audiobook-service__meta-wrapper\\"><span class=\\"card-audiobook-service__cta\\">
      Posłuchaj w
    </span> <span class=\\"card-audiobook-service__name\\">Apple Podcasts</span></div>
    </a></li>
  <li class=\\"list-audiobook-services__item\\"><a data-test=\\"components-card-audiobook-service\\" href=\\"https://169cm.pl/audiobooki/castbox\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"card-audiobook-service\\">
      <div class=\\"card-audiobook-service__logo-wrapper\\"><img alt=\\"Ikona serwisu Castbox\\" data-test=\\"components-card-audiobook-service-logo\\" src=\\"castbox.svg\\" class=\\"card-audiobook-service__logo\\"></div>
      <div class=\\"card-audiobook-service__meta-wrapper\\"><span class=\\"card-audiobook-service__cta\\">
      Posłuchaj w
    </span> <span class=\\"card-audiobook-service__name\\">Castbox</span></div>
    </a></li>
  <li class=\\"list-audiobook-services__item\\"><a data-test=\\"components-card-audiobook-service\\" href=\\"https://169cm.pl/audiobooki/google-podcasts\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"card-audiobook-service\\">
      <div class=\\"card-audiobook-service__logo-wrapper\\"><img alt=\\"Ikona serwisu Google Podcasts\\" data-test=\\"components-card-audiobook-service-logo\\" src=\\"google-podcasts.svg\\" class=\\"card-audiobook-service__logo\\"></div>
      <div class=\\"card-audiobook-service__meta-wrapper\\"><span class=\\"card-audiobook-service__cta\\">
      Posłuchaj w
    </span> <span class=\\"card-audiobook-service__name\\">Google Podcasts</span></div>
    </a></li>
  <li class=\\"list-audiobook-services__item\\"><a data-test=\\"components-card-audiobook-service\\" href=\\"https://169cm.pl/audiobooki/pocket-casts\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"card-audiobook-service\\">
      <div class=\\"card-audiobook-service__logo-wrapper\\"><img alt=\\"Ikona serwisu Pocket Casts\\" data-test=\\"components-card-audiobook-service-logo\\" src=\\"pocket-casts.svg\\" class=\\"card-audiobook-service__logo\\"></div>
      <div class=\\"card-audiobook-service__meta-wrapper\\"><span class=\\"card-audiobook-service__cta\\">
      Posłuchaj w
    </span> <span class=\\"card-audiobook-service__name\\">Pocket Casts</span></div>
    </a></li>
  <li class=\\"list-audiobook-services__item\\"><a data-test=\\"components-card-audiobook-service\\" href=\\"https://169cm.pl/audiobooki/soundcloud\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"card-audiobook-service\\">
      <div class=\\"card-audiobook-service__logo-wrapper\\"><img alt=\\"Ikona serwisu Soundcloud\\" data-test=\\"components-card-audiobook-service-logo\\" src=\\"soundcloud.svg\\" class=\\"card-audiobook-service__logo\\"></div>
      <div class=\\"card-audiobook-service__meta-wrapper\\"><span class=\\"card-audiobook-service__cta\\">
      Posłuchaj w
    </span> <span class=\\"card-audiobook-service__name\\">Soundcloud</span></div>
    </a></li>
  <li class=\\"list-audiobook-services__item\\"><a data-test=\\"components-card-audiobook-service\\" href=\\"https://169cm.pl/audiobooki/spotify\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"card-audiobook-service\\">
      <div class=\\"card-audiobook-service__logo-wrapper\\"><img alt=\\"Ikona serwisu Spotify\\" data-test=\\"components-card-audiobook-service-logo\\" src=\\"spotify.svg\\" class=\\"card-audiobook-service__logo\\"></div>
      <div class=\\"card-audiobook-service__meta-wrapper\\"><span class=\\"card-audiobook-service__cta\\">
      Posłuchaj w
    </span> <span class=\\"card-audiobook-service__name\\">Spotify</span></div>
    </a></li>
  <li class=\\"list-audiobook-services__item\\"><a data-test=\\"components-card-audiobook-service\\" href=\\"https://169cm.pl/audiobooki/youtube\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\" class=\\"card-audiobook-service\\">
      <div class=\\"card-audiobook-service__logo-wrapper\\"><img alt=\\"Ikona serwisu YouTube\\" data-test=\\"components-card-audiobook-service-logo\\" src=\\"youtube.svg\\" class=\\"card-audiobook-service__logo\\"></div>
      <div class=\\"card-audiobook-service__meta-wrapper\\"><span class=\\"card-audiobook-service__cta\\">
      Posłuchaj w
    </span> <span class=\\"card-audiobook-service__name\\">YouTube</span></div>
    </a></li>
</ul>"
`;

exports[`Blocks / List Audiobook Services Unit shall pass unit sanity test 1`] = `
"<ul data-test=\\"blocks-list-audiobook-services\\" class=\\"list-audiobook-services\\">
  <li class=\\"list-audiobook-services__item\\">
    <componentscardaudiobookservice-stub icon=\\"apple-podcasts.svg\\" name=\\"Apple Podcasts\\" url=\\"https://169cm.pl/audiobooki/itunes\\"></componentscardaudiobookservice-stub>
  </li>
  <li class=\\"list-audiobook-services__item\\">
    <componentscardaudiobookservice-stub icon=\\"castbox.svg\\" name=\\"Castbox\\" url=\\"https://169cm.pl/audiobooki/castbox\\"></componentscardaudiobookservice-stub>
  </li>
  <li class=\\"list-audiobook-services__item\\">
    <componentscardaudiobookservice-stub icon=\\"google-podcasts.svg\\" name=\\"Google Podcasts\\" url=\\"https://169cm.pl/audiobooki/google-podcasts\\"></componentscardaudiobookservice-stub>
  </li>
  <li class=\\"list-audiobook-services__item\\">
    <componentscardaudiobookservice-stub icon=\\"pocket-casts.svg\\" name=\\"Pocket Casts\\" url=\\"https://169cm.pl/audiobooki/pocket-casts\\"></componentscardaudiobookservice-stub>
  </li>
  <li class=\\"list-audiobook-services__item\\">
    <componentscardaudiobookservice-stub icon=\\"soundcloud.svg\\" name=\\"Soundcloud\\" url=\\"https://169cm.pl/audiobooki/soundcloud\\"></componentscardaudiobookservice-stub>
  </li>
  <li class=\\"list-audiobook-services__item\\">
    <componentscardaudiobookservice-stub icon=\\"spotify.svg\\" name=\\"Spotify\\" url=\\"https://169cm.pl/audiobooki/spotify\\"></componentscardaudiobookservice-stub>
  </li>
  <li class=\\"list-audiobook-services__item\\">
    <componentscardaudiobookservice-stub icon=\\"youtube.svg\\" name=\\"YouTube\\" url=\\"https://169cm.pl/audiobooki/youtube\\"></componentscardaudiobookservice-stub>
  </li>
</ul>"
`;
